All entities classes with hbm mapping

Mononofuu opened the pull request.

skyfenko commented on core/src/main/java/edu/softserve/zoo/model/Animal.java:
```diff
@@ -0,0 +1,69 @@
+    private Species species;
+    private House house;
+    private LocalDate birthday;
+    private int temperatureMin;
```
let's assume this field is empty in DB. which value will it be applied for this field? replace with wrapper type

skyfenko commented on core/src/main/java/edu/softserve/zoo/model/Class.java:
```diff
@@ -0,0 +1,5 @@
+package edu.softserve.zoo.model;
+
+public class Class extends NamedEntity {
```
change name of class. only one class with name Class should be in java

skyfenko commented on core/src/main/java/edu/softserve/zoo/model/GeographicZone.java:
```diff
@@ -0,0 +1,29 @@
+        this.value = value;
+    }
+
+    public enum Value {
```
Value doesn't give enough clarity to identify what it stands for

skyfenko commented on core/src/main/java/edu/softserve/zoo/model/House.java:
```diff
@@ -0,0 +1,22 @@
+
+public class House extends BaseEntity {
+    private ZooZone zone;
+    private int maxCapacity;
```
think of Integer instead of int

skyfenko commented on core/src/main/java/edu/softserve/zoo/model/NamedEntity.java:
```diff
@@ -0,0 +1,26 @@
+
+    @Override
+    public String toString() {
+        return name;
```
it doesn't make sense if you have toString in all classes down

skyfenko commented on persistence/src/main/resources/hbm/GeographicZone.hbm.xml:
```diff
@@ -0,0 +1,15 @@
+                    table="geo_zones">
+        <property name="value" column="name">
+            <type name="org.hibernate.type.EnumType">
+                <param name="enumClass">edu.softserve.zoo.model.GeographicZone$Value</param>
```
are you sure that $ mark would work here?

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/Employee.java:
```diff
@@ -0,0 +1,94 @@
+                ", lastName='" + lastName + '\'' +
+                ", employedFrom=" + employedFrom +
+                ", email='" + email + '\'' +
+                ", password='" + password + '\'' +
```
nice

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/Employee.java:
```diff
@@ -0,0 +1,94 @@
+                "id=" + id +
+                ", firstName='" + firstName + '\'' +
+                ", lastName='" + lastName + '\'' +
+                ", employedFrom=" + employedFrom +
```
employmentDate

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/BaseEntity.java:
```diff
@@ -0,0 +1,36 @@
+        this.id = id;
+    }
+
+    @Override
```
no sense in equals and hc

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/GeographicalZone.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    public enum Region {
+        NORTH_AMERICA,
```
ordinal if needed

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/GeographicalZone.java:
```diff
@@ -0,0 +1,56 @@
+        this.region = region;
+    }
+
+    public enum Region {
```
no need

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/NamedEntity.java:
```diff
@@ -0,0 +1,40 @@
+package edu.softserve.zoo.model;
```
no need

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/Role.java:
```diff
@@ -0,0 +1,44 @@
+    private Type type;
+
+    public enum Type {
+        ADMIN, MANAGER, EMPLOYEE
```
ordinal

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/Species.java:
```diff
@@ -0,0 +1,74 @@
+
+public class Species extends BaseEntity {
+    private Family family;
+    private String longName;
```
latin name

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/Task.java:
```diff
@@ -0,0 +1,134 @@
+    private Employee assignee;
+    private Employee assigner;
+    private LocalDateTime plannedStart;
+    private LocalDateTime plannedFinish;
```
estimated

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/TaskStatus.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    public enum Status {
+        IN_PROGRESS, TO_DO, DONE, FAILED
```
ordinal

ITsvetkoFF commented on core/src/main/java/edu/softserve/zoo/model/TaskType.java:
```diff
@@ -0,0 +1,44 @@
+package edu.softserve.zoo.model;
+
+public class TaskType extends BaseEntity {
```
can be in task

ITsvetkoFF commented on persistence/src/main/resources/hibernate.cfg.xml:
```diff
@@ -0,0 +1,23 @@
+    "http://www.hibernate.org/dtd/hibernate-configuration-3.0.dtd">
+<hibernate-configuration>
+  <session-factory>
+    <mapping resource="hbm/BaseEntity.hbm.xml"/>
```
all in one

Mononofuu committed:
Refactoring:
 - Named Entity removed
 - TaskStatus and TaskType moved into Task class
 - Hibernate mapping reworked

skyfenko commented on core/src/main/java/edu/softserve/zoo/model/BaseEntity.java:
```diff
@@ -0,0 +1,20 @@
+package edu.softserve.zoo.model;
+
+public abstract class BaseEntity {
+    protected Integer id;
```
why do you need protected field if you have accessor and mutator?
